src/api/Feed.js

```diff
@@ -173,28 +173,32 @@ class Feed extends Base {
             }
         }
 
-        this.id = id;
+        this.file = file;
         this.hasError = false;
         this.errorCallback = onError;
         const versionsPromise = this.fetchVersions();
+        const currentVersionPromise = this.fetchCurrentVersion();
         const commentsPromise = this.fetchComments(permissions);
         const tasksPromise = shouldShowNewTasks ? this.fetchTasksNew() : this.fetchTasks();
         const appActivityPromise = shouldShowAppActivity ? this.fetchAppActivity(permissions) : Promise.resolve();
 
-        Promise.all([versionsPromise, commentsPromise, tasksPromise, appActivityPromise]).then(feedItems => {
-            const versions: ?FileVersions = feedItems[0];
-            const versionsWithRestoredVersion = this.versionsAPI.addCurrentVersion(versions, file);
-            const unsortedFeedItems = [versionsWithRestoredVersion, ...feedItems.slice(1)];
-            const sortedFeedItems = sortFeedItems(...unsortedFeedItems);
-            if (!this.isDestroyed()) {
-                this.setCachedItems(id, sortedFeedItems);
-                if (this.hasError) {
-                    errorCallback(sortedFeedItems);
-                } else {
-                    successCallback(sortedFeedItems);
+        Promise.all([versionsPromise, currentVersionPromise, commentsPromise, tasksPromise, appActivityPromise]).then(
+            ([versions: ?FileVersions, currentVersion: ?BoxItemVersion, ...feedItems]) => {
+                const decoratedVersion =
+                    currentVersion && versions
```

Comment:
Can this be refined a bit to avoid the ternary?

Reply:
yep, will revisit